### PR TITLE
Handling errors feature

### PR DESCRIPTION
**Pull request by awnigharbia**
- created a custom error message to be shown when isError indicator is true which is returned from react-query
- return Promise.reject from try, catch so react-query can catch errors

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        with:
+          path: ${{steps.yarn-cache.outputs.dir}}
+          key: ${{runner.os}}--yarn-${{hashFiles('**/yarn.lock')}}
+          restore-keys: | ${{runner.os}}--yarn-
+
+      - name: Installing dependencies
+        run: yarn install
+
+      - name: Building App
+        run: yarn build
```
Comment (awnigharbia) [inline .github/workflows/publish.yaml:34]: Changed to `npm` because we are not using `yarn`

File: package.json [hunk 2]
```diff
@@ -4,6 +4,7 @@
   "private": true,
   "main": "src/index.tsx",
   "dependencies": {
+    "@babel/core": "^7.15.5",
```
Comment (awnigharbia) [inline package.json:7]: No used. Deleted.

File: package.json [hunk 3]
```diff
@@ -48,6 +49,7 @@
     ]
   },
   "devDependencies": {
+    "@types/babel__core": "^7.1.16",
```
Comment (awnigharbia) [inline package.json:52]: No used. Deleted.

File: src/api/movieAPI.ts [hunk 4]
```diff
@@ -186,22 +177,28 @@ export function useMovieSearch(movieTitle: string): UseMoviesQueryResult {
...
+    return Promise.reject(error)
   }
 }
 
 type UseMovieQueryResult = UseQueryResult<Movie | undefined> & {
   movie: Movie | undefined
 }
 
-export function useMovie(movieId: number): UseMovieQueryResult {
+export function useMovie(movieId: number | null): UseMovieQueryResult {
```
Comment (awnigharbia) [inline src/api/movieAPI.ts:201]: Revert `null` to `number` and made modalContext to return movieId number.

File: src/components/generic/movieModal/modalContent.tsx [hunk 5]
```diff
@@ -26,18 +26,18 @@ export const MovieModal: FC = () => {
...
   const {isOpen, setIsOpen} = useModal()
 
   if (isIdle) {
     return null
   }
 
   if (isFetching) {
-    return <div>Loading...</div>
+    return null
   }
```
Comment (awnigharbia) [inline src/components/generic/movieModal/modalContent.tsx:38]: Merged together

**Review by john-d-pelingo — APPROVED** [review]

**Pull request merged**